### PR TITLE
feat: metric attribute autocomplete for the aggregation type

### LINKED ISSUE #2239: (metrics): metric suggestion API for the aggregation type
This is already there since all the metrics are numeric values. Review and close if there is nothing new to be done.

### PR DESCRIPTION
**Pull request by srikanthccv**
Fixes #2239

### CODE REVIEW

File: pkg/query-service/app/clickhouseReader/reader.go
Comment (nityanandagohain) [inline pkg/query-service/app/clickhouseReader/reader.go:3685]: same as above

File: pkg/query-service/app/clickhouseReader/reader.go
Comment (srikanthccv) [reply pkg/query-service/app/clickhouseReader/reader.go]: I wasn't really sure about this here. This could be as detailed as a bit included exact type from the DB or simply the number. The client doesn't need to bother about that because it's all `number`s for them. Used the constants defined earlier for data type but added the new constants for type. What do you think about the data type? Do you see a case where it needs to really explicit than simple "number"?

Comment (nityanandagohain) [reply pkg/query-service/app/clickhouseReader/reader.go]: For logs and traces we are going with simple "number" . I think that should be good.

Comment (srikanthccv) [reply pkg/query-service/app/clickhouseReader/reader.go]: It's fine for the metrics as well.

Comment (nityanandagohain) [reply pkg/query-service/app/clickhouseReader/reader.go]: Cool, let's go with three constants then `string`, `number`,  `bool` . cc @makeavish

**Review by nityanandagohain — APPROVED** [review]

**Review by makeavish — APPROVED** [review]

**Comment by srikanthccv** [comment]
@ankitnayan this needs your approval since it touches the ee folder.

**Comment by ankitnayan** [comment]
> @ankitnayan this needs your approval since it touches the ee folder.

Is this just a request to double check or a restriction due to code owners? Suggest me a way to bypass me. It should not be blocked pending my approval

**Comment by srikanthccv** [comment]
It's a codeowners restriction. The two approvals don't count for merging eligibility since there is a one line change in ee folder. Let me check what we can do about this.

**Pull request merged**
